sql/analyzer/resolve_functions.go

```diff
@@ -49,6 +49,18 @@ func resolveTableFunctions(ctx *sql.Context, a *Analyzer, n sql.Node, _ *Scope,
 			database = privilegedDatabase.Unwrap()
 		}
 
+		var hasBindVarArgs bool
```

Comment:
Ideally we would do at least some of the work upfront to frontload/prepare analyzer logic. This makes this rule more expensive and we run it a whole second time. Not the end of the world but we can't find and swap table functions in `ApplyBindings`? Or were error tests failing because of invalid argument types? In which case the table function resolve and semantic validation should maybe be split into different rules.

Reply:
I'm not sure what you mean by swap table functions?
The tests were failing because of invalid argument types (i think); the BindVars never get replaced, so we try to evaluate them during pre, leading to panics.

Reply:
whoops meant arguments, can we not go and replace the arguments in ApplyBindings

Reply:
I do in the code I added in `plan/bindvar.go`

I think this does the same thing as the DeferredTable; the first pass leaves it unresolved, and it should work as expected in the second pass.